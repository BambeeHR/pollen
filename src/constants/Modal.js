const Modal = Object.freeze({
  ActionAlignments: {
    CENTER: 'center',
    JUSTIFY: 'justify',
    RIGHT: 'right',
  },
  Alerts: {
    SUCCESS: 'success',
    WARNING: 'warning',
    ERROR: 'error',
  },
  Headers: {
    TRANSPARENT: 'transparent',
    PRIMARY: 'primary',
    SECONDARY: 'secondary',
    WARNING: 'warning',
    ERROR: 'error',
    PRIMARY_ORNATE: 'primary-ornate',
    WARNING_ORNATE: 'warning-ornate',
    ERROR_ORNATE: 'error-ornate',
  },
  Overlays: {
    PRIMARY: 'primary',
    DARK: 'dark',
    LIGHT: 'light',
  },
  Sizes: {
    TINY: 'tiny',
    SMALL: 'small',
    REGULAR: 'regular',
    MEDIUM: 'medium',
    LARGE: 'large',
    GIANT: 'giant',
  },
});

export default Modal;
