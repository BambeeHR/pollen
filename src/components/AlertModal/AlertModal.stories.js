import { action } from '@storybook/addon-actions';
import { select, text } from '@storybook/addon-knobs';
import BaseButton from '../BaseButton/BaseButton.vue';
import AlertModal from './AlertModal.vue';
import Modal from '../../constants/Modal';

export default {
  title: 'AlertModal',
  component: AlertModal,
  parameters: {
    componentSubtitle: '',
  },
};

export const Default = () => ({
  components: { BaseButton, AlertModal },
  props: {
    cancelLabel: {
      default: text('Cancel label', 'Cancel'),
    },
    confirmLabel: {
      default: text('Confirm label', 'Continue'),
    },
    message: {
      default: text(
        'Message',
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
      ),
    },
    title: {
      default: text('Title', 'What a success!'),
    },
    variant: {
      default: select(
        'Variant',
        Object.values(Modal.Alerts),
        Modal.Alerts.SUCCESS
      ),
    },
  },
  data() {
    return { isModalOpen: true };
  },
  methods: {
    handleCancel: action('cancel'),
    handleConfirm: action('confirm'),
  },
  template: `
  <div>
    <BaseButton @click="isModalOpen = !isModalOpen">Toggle Modal</BaseButton>
    <AlertModal
      v-if="isModalOpen"
      :cancel-label="cancelLabel"
      :confirm-label="confirmLabel"
      :variant="variant"
      :title="title"
      :message="message"
      @close="isModalOpen = false"
      @cancel="handleCancel"
      @confirm="handleConfirm"
    />
  </div>
  `,
});

export const WithContentAsSlots = () => ({
  components: { BaseButton, AlertModal },
  props: {
    cancelLabel: {
      default: text('Cancel label', 'Cancel'),
    },
    confirmLabel: {
      default: text('Confirm label', 'Continue'),
    },
    message: {
      default: text(
        'Message',
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.'
      ),
    },
    title: {
      default: text('Title', 'What a success!'),
    },
    variant: {
      default: select(
        'Variant',
        Object.values(Modal.Alerts),
        Modal.Alerts.SUCCESS
      ),
    },
  },
  data() {
    return { isModalOpen: true };
  },
  methods: {
    handleCancel: action('cancel'),
    handleConfirm: action('confirm'),
  },
  template: `
  <div>
    <BaseButton @click="isModalOpen = !isModalOpen">Toggle Modal</BaseButton>
    <AlertModal
      v-if="isModalOpen"
      :cancel-label="cancelLabel"
      :confirm-label="confirmLabel"
      :variant="variant"
      @cancel="handleCancel"
      @confirm="handleConfirm"
      @close="isModalOpen = false">
      <template #header>
        {{ title }}
      </template>
      <b class="font-bold">Test</b>: {{ message }}
    </AlertModal>
  </div>
  `,
});
