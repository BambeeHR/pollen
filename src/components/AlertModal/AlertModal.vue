<template>
  <ModalDialog
    class="alert-modal"
    :class="`alert-modal--${variant}`"
    :actions="actions"
    v-bind="dialogProps"
    @close="$emit('close')"
  >
    <template #header>
      <BaseIcon class="alert-modal__icon" :icon="headerIcon" />
      <slot name="header">
        {{ title }}
      </slot>
    </template>
    <slot>
      {{ message }}
    </slot>
  </ModalDialog>
</template>

<script>
import BaseIcon, { Icons } from '../BaseIcon/BaseIcon.vue';
import ModalDialog from '../ModalDialog/ModalDialog.vue';
import Modal from '../../constants/Modal';

/**
 * import { AlertModal } from '@bambee/pollen';
 *
 * An alert modal. This will display a `success`, `warning`, or `error` dialog
 * with proper color and iconography. The modal actions will emit either a
 * `cancel` or `confirm` event.
 */
export default {
  components: { BaseIcon, ModalDialog },
  inheritAttrs: false,
  props: {
    /** The label for the cancel button. */
    cancelLabel: {
      type: String,
      default: 'Cancel',
    },
    /** The label for the confirm button. */
    confirmLabel: {
      type: String,
      default: 'Continue',
    },
    /** Message to display. This can also be defined by the `default` slot. */
    message: {
      type: String,
      default: null,
    },
    /** The title to display. This can also be defined by the `header` slot. */
    title: {
      type: String,
      default: null,
    },
    /** One of `success`, `warning`, or `error`. */
    variant: {
      type: String,
      default: Modal.Alerts.SUCCESS,
      validator: (value) => Object.values(Modal.Alerts).includes(value),
    },
  },
  computed: {
    actions() {
      return {
        primary: {
          label: this.confirmLabel,
          handler: () => {
            this.$emit('confirm');
          },
        },
        secondary: {
          label: this.cancelLabel,
          handler: () => {
            this.$emit('cancel');
          },
        },
      };
    },
    dialogProps() {
      return {
        actionAlignment: Modal.ActionAlignments.CENTER,
        header: Modal.Headers.TRANSPARENT,
        size: Modal.Sizes.SMALL,
        ...this.$attrs,
      };
    },
    headerIcon() {
      switch (this.variant) {
        case Modal.Alerts.WARNING:
          return Icons.ALERT_TRIANGLE;
        case Modal.Alerts.ERROR:
          return Icons.ALERT_CIRCLE;
        case Modal.Alerts.SUCCESS:
        default:
          return Icons.CHECK_CIRCLE;
      }
    },
  },
};
</script>

<style scoped>
.alert-modal__icon {
  @apply block mb-4 mt-2 mx-auto text-33;
}

.alert-modal--success .alert-modal__icon {
  @apply text-success;
}

.alert-modal--success >>> .modal-actions__action--primary {
  @apply bg-success;
}

.alert-modal--warning .alert-modal__icon {
  @apply text-warning;
}

.alert-modal--warning >>> .modal-actions__action--primary {
  @apply bg-warning;
}

.alert-modal--error .alert-modal__icon {
  @apply text-error;
}

.alert-modal--error >>> .modal-actions__action--primary {
  @apply bg-error;
}
</style>
