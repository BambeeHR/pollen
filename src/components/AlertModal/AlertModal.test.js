import { mount } from '@vue/test-utils';
import AlertModal from './AlertModal.vue';
import Modal from '../../constants/Modal';

describe('AlertModal', () => {
  Object.values(Modal.Alerts).forEach((variant) => {
    test(`renders ${variant} variant correctly`, () => {
      const wrapper = mount(AlertModal, {
        slots: {
          default: 'Content goes here',
          header: 'Heading goes here',
        },
        propsData: { variant },
      });
      expect(wrapper.element).toMatchSnapshot();
    });
  });

  test('emits a confirm and cancel event', () => {
    const wrapper = mount(AlertModal);
    wrapper.find('.modal-actions__action--primary').trigger('click');
    expect(wrapper.emitted().confirm.length).toBe(1);
    wrapper.find('.modal-actions__action--secondary').trigger('click');
    expect(wrapper.emitted().cancel.length).toBe(1);
  });
});
