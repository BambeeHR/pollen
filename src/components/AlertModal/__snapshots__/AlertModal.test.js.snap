// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertModal renders error variant correctly 1`] = `
<div
  class="modal-dialog alert-modal alert-modal--error"
>
  <div
    tabindex="-1"
  />
   
  <div
    tabindex="-1"
  />
   
  <div
    data-lock=""
  >
    <div
      class="modal-window modal-window--small"
    >
      <div
        class="modal-header modal-header--transparent"
      >
        <svg
          aria-hidden="true"
          class="base-icon svg-inline--fa fa-exclamation-circle fa-w-16 alert-modal__icon"
          data-icon="exclamation-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class=""
            d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
          />
        </svg>
         Heading goes here
      </div>
       
      <div
        class="modal-dialog__body"
      >
         Content goes here
      </div>
       
      <div
        class="modal-actions modal-actions--center"
      >
        <button
          class="modal-actions__action modal-actions__action--primary modal-actions__action modal-actions__action--primary base-button base-button--primary base-button--medium base-button--regular base-button--inline"
          type="button"
        >
          <div
            class="base-button__inner"
          >
            Continue
          </div>
        </button>
         
        <button
          class="modal-actions__action modal-actions__action--secondary modal-actions__action modal-actions__action--secondary base-button base-button--secondary base-button--medium base-button--flat base-button--inline"
          type="button"
        >
          <div
            class="base-button__inner"
          >
            Cancel
          </div>
        </button>
         
        <!---->
      </div>
       
      <!---->
    </div>
     
    <button
      class="modal-close modal-close--light"
      title="Close"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="base-icon svg-inline--fa fa-times fa-w-10"
        data-icon="times"
        data-prefix="fal"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
          fill="currentColor"
        />
      </svg>
      <div
        aria-hidden="true"
        class="modal-close__visible-label"
      >
        Esc
      </div>
      <div
        class="modal-close__accessible-label"
      >
        Click or press escape to close
      </div>
    </button>
     
    <div
      class="modal-overlay modal-overlay--primary"
    />
  </div>
   
  <div
    tabindex="-1"
  />
</div>
`;

exports[`AlertModal renders success variant correctly 1`] = `
<div
  class="modal-dialog alert-modal alert-modal--success"
>
  <div
    tabindex="-1"
  />
   
  <div
    tabindex="-1"
  />
   
  <div
    data-lock=""
  >
    <div
      class="modal-window modal-window--small"
    >
      <div
        class="modal-header modal-header--transparent"
      >
        <svg
          aria-hidden="true"
          class="base-icon svg-inline--fa fa-check-circle fa-w-16 alert-modal__icon"
          data-icon="check-circle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class=""
            d="M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z"
            fill="currentColor"
          />
        </svg>
         Heading goes here
      </div>
       
      <div
        class="modal-dialog__body"
      >
         Content goes here
      </div>
       
      <div
        class="modal-actions modal-actions--center"
      >
        <button
          class="modal-actions__action modal-actions__action--primary modal-actions__action modal-actions__action--primary base-button base-button--primary base-button--medium base-button--regular base-button--inline"
          type="button"
        >
          <div
            class="base-button__inner"
          >
            Continue
          </div>
        </button>
         
        <button
          class="modal-actions__action modal-actions__action--secondary modal-actions__action modal-actions__action--secondary base-button base-button--secondary base-button--medium base-button--flat base-button--inline"
          type="button"
        >
          <div
            class="base-button__inner"
          >
            Cancel
          </div>
        </button>
         
        <!---->
      </div>
       
      <!---->
    </div>
     
    <button
      class="modal-close modal-close--light"
      title="Close"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="base-icon svg-inline--fa fa-times fa-w-10"
        data-icon="times"
        data-prefix="fal"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
          fill="currentColor"
        />
      </svg>
      <div
        aria-hidden="true"
        class="modal-close__visible-label"
      >
        Esc
      </div>
      <div
        class="modal-close__accessible-label"
      >
        Click or press escape to close
      </div>
    </button>
     
    <div
      class="modal-overlay modal-overlay--primary"
    />
  </div>
   
  <div
    tabindex="-1"
  />
</div>
`;

exports[`AlertModal renders warning variant correctly 1`] = `
<div
  class="modal-dialog alert-modal alert-modal--warning"
>
  <div
    tabindex="-1"
  />
   
  <div
    tabindex="-1"
  />
   
  <div
    data-lock=""
  >
    <div
      class="modal-window modal-window--small"
    >
      <div
        class="modal-header modal-header--transparent"
      >
        <svg
          aria-hidden="true"
          class="base-icon svg-inline--fa fa-exclamation-triangle fa-w-18 alert-modal__icon"
          data-icon="exclamation-triangle"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class=""
            d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
            fill="currentColor"
          />
        </svg>
         Heading goes here
      </div>
       
      <div
        class="modal-dialog__body"
      >
         Content goes here
      </div>
       
      <div
        class="modal-actions modal-actions--center"
      >
        <button
          class="modal-actions__action modal-actions__action--primary modal-actions__action modal-actions__action--primary base-button base-button--primary base-button--medium base-button--regular base-button--inline"
          type="button"
        >
          <div
            class="base-button__inner"
          >
            Continue
          </div>
        </button>
         
        <button
          class="modal-actions__action modal-actions__action--secondary modal-actions__action modal-actions__action--secondary base-button base-button--secondary base-button--medium base-button--flat base-button--inline"
          type="button"
        >
          <div
            class="base-button__inner"
          >
            Cancel
          </div>
        </button>
         
        <!---->
      </div>
       
      <!---->
    </div>
     
    <button
      class="modal-close modal-close--light"
      title="Close"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="base-icon svg-inline--fa fa-times fa-w-10"
        data-icon="times"
        data-prefix="fal"
        focusable="false"
        role="img"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M193.94 256L296.5 153.44l21.15-21.15c3.12-3.12 3.12-8.19 0-11.31l-22.63-22.63c-3.12-3.12-8.19-3.12-11.31 0L160 222.06 36.29 98.34c-3.12-3.12-8.19-3.12-11.31 0L2.34 120.97c-3.12 3.12-3.12 8.19 0 11.31L126.06 256 2.34 379.71c-3.12 3.12-3.12 8.19 0 11.31l22.63 22.63c3.12 3.12 8.19 3.12 11.31 0L160 289.94 262.56 392.5l21.15 21.15c3.12 3.12 8.19 3.12 11.31 0l22.63-22.63c3.12-3.12 3.12-8.19 0-11.31L193.94 256z"
          fill="currentColor"
        />
      </svg>
      <div
        aria-hidden="true"
        class="modal-close__visible-label"
      >
        Esc
      </div>
      <div
        class="modal-close__accessible-label"
      >
        Click or press escape to close
      </div>
    </button>
     
    <div
      class="modal-overlay modal-overlay--primary"
    />
  </div>
   
  <div
    tabindex="-1"
  />
</div>
`;
